common/djangoapps/third_party_auth/strategy.py

```diff
@@ -29,16 +29,21 @@ def setting(self, name, default=None, backend=None):
             setting 'name' is configured via LTIProviderConfig.
         """
         if isinstance(backend, OAuthAuth):
+            raise_exception = False
             oauth2_slugs = OAuth2ProviderConfig.key_values('provider_slug', flat=True)
             for oauth2_slug in oauth2_slugs:
                 provider = OAuth2ProviderConfig.current(oauth2_slug)
-                if provider.enabled_for_current_site and provider.backend_name == backend.name:
-                    try:
-                        return provider.get_setting(name)
-                    except KeyError:
-                        pass
-                elif not provider.enabled_for_current_site and provider.backend_name == backend.name:
-                    raise Exception("Can't fetch setting of a disabled backend/provider.")
+                if provider.backend_name == backend.name:
```

Comment:
Just to know, what happen if provider.backend_name is different than backend.name?

Reply:
Nothing happens, this resumes the iteration